Panic on 25.4.1

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

### What happened?

Updated to 25.4.1 and get this panic fairly frequently.  Easiest way for me to reproduce is it to spam enter 

fatal error: concurrent map iteration and map write

goroutine 1 [running]:
internal/runtime/maps.fatal({0x100a44a6d?, 0x140000ae598?})
        /opt/homebrew/Cellar/go/1.24.1/libexec/src/runtime/panic.go:1058 +0x20
internal/runtime/maps.(*Iter).Next(0x140002c70e8?)
        /opt/homebrew/Cellar/go/1.24.1/libexec/src/internal/runtime/maps/table.go:683 +0x94
github.com/jandedobbeleer/oh-my-posh/src/template.(*fields).init(0x140002c75e0, {0x100cae8c0, 0x140000ba150})
        /private/tmp/oh-my-posh-20250312-97719-nklkpr/oh-my-posh-25.4.1/src/template/text.go:140 +0x148
github.com/jandedobbeleer/oh-my-posh/src/template.(*fields).init(0x140002c75e0, {0x100d065e0, 0x140000320c0})
        /private/tmp/oh-my-posh-20250312-97719-nklkpr/oh-my-posh-25.4.1/src/template/text.go:161 +0x2e4
github.com/jandedobbeleer/oh-my-posh/src/template.(*fields).init(0x140002c75e0, {0x100d10e80, 0x1400030e000})
        /private/tmp/oh-my-posh-20250312-97719-nklkpr/oh-my-posh-25.4.1/src/template/text.go:182 +0x1d4
github.com/jandedobbeleer/oh-my-posh/src/template.(*Text).patchTemplate(0x140002c7748)
        /private/tmp/oh-my-posh-20250312-97719-nklkpr/oh-my-posh-25.4.1/src/template/text.go:34 +0x3c
github.com/jandedobbeleer/oh-my-posh/src/template.(*Text).Render(0x140002c7748)
        /private/tmp/oh-my-posh-20250312-97719-nklkpr/oh-my-posh-25.4.1/src/template/text.go:24 +0xd0
github.com/jandedobbeleer/oh-my-posh/src/config.(*Segment).string(0x140002a6400)
        /private/tmp/oh-my-posh-20250312-97719-nklkpr/oh-my-posh-25.4.1/src/config/segment.go:324 +0xdc
github.com/jandedobbeleer/oh-my-posh/src/config.(*Segment).Render(0x140002a6400, 0x140001a7050?)
        /private/tmp/oh-my-posh-20250312-97719-nklkpr/oh-my-posh-25.4.1/src/config/segment.go:155 +0x40
github.com/jandedobbeleer/oh-my-posh/src/prompt.(*Engine).writeSegments(0x140001a7290, 0x140002fa8c0, 0x140001a7050)
        /private/tmp/oh-my-posh-20250312-97719-nklkpr/oh-my-posh-25.4.1/src/prompt/segments.go:80 +0x2ac
github.com/jandedobbeleer/oh-my-posh/src/prompt.(*Engine).writeBlockSegments(0x140001a7290, 0x140001a7050)
        /private/tmp/oh-my-posh-20250312-97719-nklkpr/oh-my-posh-25.4.1/src/prompt/segments.go:32 +0x154
github.com/jandedobbeleer/oh-my-posh/src/prompt.(*Engine).renderBlock(0x140001a7290, 0x140001a7050, 0x0)
        /private/tmp/oh-my-posh-20250312-97719-nklkpr/oh-my-posh-25.4.1/src/prompt/engine.go:197 +0x3c
github.com/jandedobbeleer/oh-my-posh/src/prompt.(*Engine).writePrimaryPrompt(0x140001a7290, 0x1)
        /private/tmp/oh-my-posh-20250312-97719-nklkpr/oh-my-posh-25.4.1/src/prompt/primary.go:72 +0x410
github.com/jandedobbeleer/oh-my-posh/src/prompt.(*Engine).Primary(0x140001a7290)
        /private/tmp/oh-my-posh-20250312-97719-nklkpr/oh-my-posh-25.4.1/src/prompt/primary.go:15 +0x30
github.com/jandedobbeleer/oh-my-posh/src/cli.createPrintCmd.func1(0x140002c8308, {0x140002cf0e0, 0x4?, 0x100a1bccf?})
        /private/tmp/oh-my-posh-20250312-97719-nklkpr/oh-my-posh-25.4.1/src/cli/print.go:95 +0x444
github.com/spf13/cobra.(*Command).execute(0x140002c8308, {0x140002cefa0, 0xa, 0xa})
        /Users/markohara/Library/Caches/Homebrew/go_mod_cache/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1019 +0x810
github.com/spf13/cobra.(*Command).ExecuteC(0x10126efe0)
        /Users/markohara/Library/Caches/Homebrew/go_mod_cache/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1148 +0x350
github.com/spf13/cobra.(*Command).Execute(...)
        /Users/markohara/Library/Caches/Homebrew/go_mod_cache/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1071
github.com/jandedobbeleer/oh-my-posh/src/cli.Execute()
        /private/tmp/oh-my-posh-20250312-97719-nklkpr/oh-my-posh-25.4.1/src/cli/root.go:46 +0x24
main.main()
        /private/tmp/oh-my-posh-20250312-97719-nklkpr/oh-my-posh-25.4.1/src/main.go:8 +0x1c

goroutine 26 [runnable]:
reflect.(*rtype).String(0x100c2f180)
        /opt/homebrew/Cellar/go/1.24.1/libexec/src/reflect/type.go:569 +0x84
reflect.stringFor(...)
        /opt/homebrew/Cellar/go/1.24.1/libexec/src/reflect/type.go:1932
reflect.typesByString({0x14000164500, 0x42})
        /opt/homebrew/Cellar/go/1.24.1/libexec/src/reflect/type.go:1702 +0x13c
reflect.FuncOf({0x140002df5f0, 0x3, 0x0?}, {0x1013ae0a0, 0x0, 0x0?}, 0x0)
        /opt/homebrew/Cellar/go/1.24.1/libexec/src/reflect/type.go:1920 +0x53c
reflect.(*rtype).Method(0x100cbece0, 0x1)
        /opt/homebrew/Cellar/go/1.24.1/libexec/src/reflect/type.go:636 +0x3c4
github.com/jandedobbeleer/oh-my-posh/src/template.(*fields).init(0x1400004bc20, {0x100cdc480, 0x14000119680})
        /private/tmp/oh-my-posh-20250312-97719-nklkpr/oh-my-posh-25.4.1/src/template/text.go:169 +0x328
github.com/jandedobbeleer/oh-my-posh/src/template.(*fields).init(0x14000088c20, {0x100cbece0, 0x14000119620})
        /private/tmp/oh-my-posh-20250312-97719-nklkpr/oh-my-posh-25.4.1/src/template/text.go:182 +0x1d4
github.com/jandedobbeleer/oh-my-posh/src/template.(*Text).patchTemplate(0x140002b2ce0)
        /private/tmp/oh-my-posh-20250312-97719-nklkpr/oh-my-posh-25.4.1/src/template/text.go:34 +0x3c
github.com/jandedobbeleer/oh-my-posh/src/template.(*Text).Render(0x140002b2ce0)
        /private/tmp/oh-my-posh-20250312-97719-nklkpr/oh-my-posh-25.4.1/src/template/text.go:24 +0xd0
github.com/jandedobbeleer/oh-my-posh/src/segments.(*Status).formatStatus(0x14000119620, 0x0, {0x16fb473f0, 0x1})
        /private/tmp/oh-my-posh-20250312-97719-nklkpr/oh-my-posh-25.4.1/src/segments/status.go:88 +0x36c
github.com/jandedobbeleer/oh-my-posh/src/segments.(*Status).Enabled(0x14000119620)
        /private/tmp/oh-my-posh-20250312-97719-nklkpr/oh-my-posh-25.4.1/src/segments/status.go:33 +0x44
github.com/jandedobbeleer/oh-my-posh/src/config.(*Segment).Execute(0x140002a6800, {0x100d46498, 0x140001a70e0})
        /private/tmp/oh-my-posh-20250312-97719-nklkpr/oh-my-posh-25.4.1/src/config/segment.go:126 +0x2a8
github.com/jandedobbeleer/oh-my-posh/src/prompt.(*Engine).writeBlockSegments.func1(0x140002a6800)
        /private/tmp/oh-my-posh-20250312-97719-nklkpr/oh-my-posh-25.4.1/src/prompt/segments.go:27 +0x3c
created by github.com/jandedobbeleer/oh-my-posh/src/prompt.(*Engine).writeBlockSegments in goroutine 1
        /private/tmp/oh-my-posh-20250312-97719-nklkpr/oh-my-posh-25.4.1/src/prompt/segments.go:26 +0x7c


### Theme

{
  "$schema": "https://raw.githubusercontent.com/JanDeDobbeleer/oh-my-posh/main/themes/schema.json",
  "blocks": [
    {
      "type": "prompt",
      "alignment": "left",
      "segments": [
        {
          "properties": {
            "cache_duration": "none"
          },
          "template": "➜",
          "foreground": "#98C379",
          "type": "text",
          "style": "plain"
        },
        {
          "properties": {
            "cache_duration": "none",
            "style": "folder"
          },
          "template": " {{ .Path }} ",
          "foreground": "#56B6C2",
          "type": "path",
          "style": "plain"
        },
        {
          "properties": {
            "branch_icon": "",
            "cache_duration": "none"
          },
          "template": "<#5FAAE8>git:(</>{{ .HEAD }}<#5FAAE8>)</>",
          "foreground": "#D0666F",
          "type": "git",
          "style": "plain"
        },
        {
          "properties": {
            "cache_duration": "none"
          },
          "template": " ✗",
          "foreground": "#BF616A",
          "type": "status",
          "style": "plain"
        }
      ]
    }
  ],
  "version": 3,
  "final_space": true
}


### What OS are you seeing the problem on?

macOS

### Which shell are you using?

zsh

### Log output

```Shell
Version: 25.4.1

Shell: zsh (5.9)

Prompt:

➜ ~

Segments:

ConsoleTitle(false)                       -   0 ms
Text(true)                                -   0 ms
Path(true)                                -   1 ms
Git(false)                                -   0 ms
Status(false)                             -   2 ms

Run duration: 14.522416ms

Cache path: /Users/markohara/.cache/oh-my-posh

Config path: /Users/markohara/.config/oh-my-posh/theme.json

Logs:

[DEBUG] 20:30:54.652 debug.go:36 → debug mode enabled
[DEBUG] 20:30:54.652 load.go:Path:79 → config set using POSH_THEME: /Users/markohara/.config/oh-my-posh/theme.json
[TRACE] 20:30:54.652 load.go:Path() - 331.625µs
[TRACE] 20:30:54.653 load.go:loadConfig() - 802.292µs
[TRACE] 20:30:54.653 load.go:Load() - 806.792µs
[DEBUG] 20:30:54.654 terminal.go:Init:54 → plain mode enabled
[DEBUG] 20:30:54.654 home.go:12 → /Users/markohara
[TRACE] 20:30:54.654 path.go:Path() - 74.917µs
[DEBUG] 20:30:54.654 file.go:Init:26 → loading cache file: /Users/markohara/.cache/oh-my-posh/omp.cache
[DEBUG] 20:30:54.654 file.go:Init:46 → loading cache key: environment_platform
[DEBUG] 20:30:54.654 file.go:Init:46 → loading cache key: is_wsl
[DEBUG] 20:30:54.654 file.go:Init:46 → loading cache key: upgrade_check
[TRACE] 20:30:54.654 file.go:Init(/Users/markohara/.cache/oh-my-posh/omp.cache) - 259.583µs
[DEBUG] 20:30:54.654 home.go:12 → /Users/markohara
[TRACE] 20:30:54.654 path.go:Path() - 8.125µs
[DEBUG] 20:30:54.654 file.go:Init:26 → loading cache file: /Users/markohara/.cache/oh-my-posh/omp.cache.ae9e9972-306a-4fe5-a986-2cfa263d073f
[DEBUG] 20:30:54.654 file.go:Init:46 → loading cache key: prompt_count_cache
[TRACE] 20:30:54.654 file.go:Init(/Users/markohara/.cache/oh-my-posh/omp.cache.ae9e9972-306a-4fe5-a986-2cfa263d073f) - 143.417µs
[TRACE] 20:30:54.654 terminal.go:setPromptCount() - 333ns
[TRACE] 20:30:54.654 terminal.go:GOOS() - 41ns
[DEBUG] 20:30:54.654 terminal.go:setPwd:115 → /Users/markohara
[TRACE] 20:30:54.654 terminal.go:setPwd() - 26.042µs
[TRACE] 20:30:54.654 terminal.go:Init() - 1.090875ms
[TRACE] 20:30:54.654 terminal.go:Shell() - 42ns
[TRACE] 20:30:54.654 terminal.go:Flags() - 0s
[TRACE] 20:30:54.654 cache.go:restoreCache() - 291ns
[TRACE] 20:30:54.654 terminal_unix.go:Root() - 458ns
[TRACE] 20:30:54.654 terminal.go:Shell() - 42ns
[TRACE] 20:30:54.654 terminal.go:Flags() - 0s
[TRACE] 20:30:54.654 terminal.go:StatusCodes() - 41ns
[DEBUG] 20:30:54.654 terminal_unix.go:IsWsl:32 → false
[TRACE] 20:30:54.654 terminal_unix.go:IsWsl() - 2.125µs
[TRACE] 20:30:54.654 terminal.go:Flags() - 42ns
[TRACE] 20:30:54.654 terminal.go:Flags() - 0s
[DEBUG] 20:30:54.654 home.go:12 → /Users/markohara
[DEBUG] 20:30:54.654 terminal_unix.go:IsWsl:32 → false
[TRACE] 20:30:54.654 terminal_unix.go:IsWsl() - 1.834µs
[TRACE] 20:30:54.654 terminal.go:Flags() - 42ns
[TRACE] 20:30:54.654 terminal.go:Flags() - 42ns
[TRACE] 20:30:54.654 terminal.go:GOOS() - 42ns
[DEBUG] 20:30:54.654 terminal.go:User:249 → markohara
[TRACE] 20:30:54.654 terminal.go:User() - 1.791µs
[DEBUG] 20:30:54.654 terminal.go:Host:266 → Marks-MBP
[TRACE] 20:30:54.654 terminal.go:Host() - 6.667µs
[TRACE] 20:30:54.654 terminal.go:GOOS() - 41ns
[DEBUG] 20:30:54.654 terminal.go:Getenv:77 → 1
[TRACE] 20:30:54.654 terminal.go:Getenv(SHLVL) - 1.708µs
[DEBUG] 20:30:54.654 writer.go:Init:105 → terminal program: ghostty
[DEBUG] 20:30:54.654 writer.go:Init:106 → terminal shell: shell
[DEBUG] 20:30:54.655 terminal.go:Getenv:77 → NO DATA
[TRACE] 20:30:54.655 terminal.go:Getenv(OMP_CACHE_DISABLED) - 1.833µs
[ERROR] 20:30:54.662 terminal.go:RunCommand:290 → exit status 1
[DEBUG] 20:30:54.662 terminal.go:RunCommand:293 ↓
    2025-03-12 20:30:54.661 defaults[98611:2329490]
    The domain/default pair of (kCFPreferencesAnyApplication, AppleAccentColor) does not exist

[TRACE] 20:30:54.662 terminal.go:RunCommand(defaults read -g AppleAccentColor) - 7.559584ms
[ERROR] 20:30:54.662 colors_darwin.go:GetAccentColor:14 → exit status 1
[TRACE] 20:30:54.662 colors.go:SetAccentColor() - 7.581ms
[TRACE] 20:30:54.662 terminal.go:Shell() - 125ns
[DEBUG] 20:30:54.662 terminal.go:Getenv:77 → 5.9
[TRACE] 20:30:54.662 terminal.go:Getenv(POSH_SHELL_VERSION) - 4.25µs
[DEBUG] 20:30:54.662 debug.go:PrintDebug:24 → segment: Title
[TRACE] 20:30:54.662 text.go:Render() - 83ns
[TRACE] 20:30:54.662 terminal.go:Flags() - 0s
[DEBUG] 20:30:54.662 terminal.go:Getenv:77 → NO DATA
[TRACE] 20:30:54.662 terminal.go:Getenv(POSH_CURSOR_LINE) - 2.042µs
[DEBUG] 20:30:54.662 terminal.go:Getenv:77 → NO DATA
[TRACE] 20:30:54.662 terminal.go:Getenv(POSH_CURSOR_COLUMN) - 1.334µs
[TRACE] 20:30:54.662 terminal.go:Flags() - 42ns
[TRACE] 20:30:54.662 terminal.go:Flags() - 41ns
[TRACE] 20:30:54.663 terminal.go:Flags() - 41ns
[TRACE] 20:30:54.663 terminal.go:Flags() - 0s
[DEBUG] 20:30:54.663 home.go:12 → /Users/markohara
[TRACE] 20:30:54.663 terminal.go:GOOS() - 83ns
[TRACE] 20:30:54.663 terminal.go:Flags() - 83ns
[DEBUG] 20:30:54.663 home.go:12 → /Users/markohara
[TRACE] 20:30:54.663 terminal.go:GOOS() - 41ns
[DEBUG] 20:30:54.663 segment.go:Execute:111 → segment: Status
[DEBUG] 20:30:54.663 segment.go:Execute:111 → segment: Git
[DEBUG] 20:30:54.663 segment.go:isToggled:228 → no toggles found
[DEBUG] 20:30:54.663 segment.go:isToggled:228 → no toggles found
[DEBUG] 20:30:54.663 home.go:12 → /Users/markohara
[TRACE] 20:30:54.663 terminal.go:GOOS() - 42ns
[DEBUG] 20:30:54.663 segment.go:Execute:111 → segment: Path
[DEBUG] 20:30:54.663 segment.go:isToggled:228 → no toggles found
[TRACE] 20:30:54.663 terminal.go:Flags() - 83ns
[DEBUG] 20:30:54.663 home.go:12 → /Users/markohara
[TRACE] 20:30:54.663 terminal.go:GOOS() - 41ns
[DEBUG] 20:30:54.663 segment.go:Execute:111 → segment: Text
[DEBUG] 20:30:54.663 segment.go:isToggled:228 → no toggles found
[TRACE] 20:30:54.663 text.go:Render(➜) - 125ns
[TRACE] 20:30:54.663 text.go:Render(plain) - 42ns
[DEBUG] 20:30:54.663 terminal_unix.go:IsWsl:32 → false
[TRACE] 20:30:54.663 terminal_unix.go:IsWsl() - 229.875µs
[TRACE] 20:30:54.663 terminal_unix.go:IsWsl2() - 232.125µs
[TRACE] 20:30:54.663 terminal.go:StatusCodes() - 84ns
[TRACE] 20:30:54.663 terminal.go:GOOS() - 42ns
[DEBUG] 20:30:54.663 properties.go:GetString:28 → status_template: {{ .Code }}
[DEBUG] 20:30:54.663 terminal.go:CommandPath:317 → /opt/homebrew/bin/git
[TRACE] 20:30:54.663 terminal.go:CommandPath(git) - 44.833µs
[TRACE] 20:30:54.663 terminal.go:HasCommand(git) - 46.625µs
[DEBUG] 20:30:54.663 properties.go:GetBool:22 → fetch_bare_info: false
[DEBUG] 20:30:54.663 terminal.go:ResolveSymlink:208 → /Users/markohara
[TRACE] 20:30:54.663 terminal.go:ResolveSymlink(/Users/markohara) - 23.542µs
[DEBUG] 20:30:54.663 properties.go:GetBool:22 → display_cygpath: false
[TRACE] 20:30:54.663 terminal.go:GOOS() - 83ns
[TRACE] 20:30:54.663 separator.go:Separator() - 0s
[TRACE] 20:30:54.663 terminal.go:Shell() - 83ns
[TRACE] 20:30:54.663 terminal.go:Shell() - 42ns
[ERROR] 20:30:54.663 terminal.go:HasParentFilePath:476 → stat .git: no such file or directory
[TRACE] 20:30:54.663 terminal.go:HasParentFilePath(.git) - 90.208µs
[TRACE] 20:30:54.663 terminal.go:GOOS() - 41ns
[DEBUG] 20:30:54.664 properties.go:GetBool:22 → mapped_locations_enabled: true
[DEBUG] 20:30:54.664 properties.go:GetString:28 → windows_registry_icon: 
[DEBUG] 20:30:54.664 properties.go:GetString:28 → windows_registry_icon: 
[DEBUG] 20:30:54.664 home.go:12 → /Users/markohara
[TRACE] 20:30:54.664 separator.go:Separator() - 41ns
[TRACE] 20:30:54.664 terminal.go:GOOS() - 42ns
[TRACE] 20:30:54.664 terminal.go:GOOS() - 41ns
[DEBUG] 20:30:54.664 properties.go:GetString:28 → home_icon: ~
[DEBUG] 20:30:54.664 properties.go:GetKeyValueMap:46 → mapped_locations: map[]
[TRACE] 20:30:54.664 separator.go:Separator() - 0s
[TRACE] 20:30:54.664 terminal.go:GOOS() - 42ns
[TRACE] 20:30:54.664 terminal.go:GOOS() - 42ns
[TRACE] 20:30:54.664 separator.go:Separator() - 42ns
[TRACE] 20:30:54.664 terminal.go:GOOS() - 42ns
[TRACE] 20:30:54.664 terminal.go:GOOS() - 0s
[TRACE] 20:30:54.664 terminal.go:GOOS() - 0s
[TRACE] 20:30:54.664 terminal.go:GOOS() - 42ns
[TRACE] 20:30:54.664 terminal.go:GOOS() - 41ns
[TRACE] 20:30:54.664 terminal.go:Flags() - 125ns
[DEBUG] 20:30:54.664 properties.go:GetStringArray:52 → cycle: []
[DEBUG] 20:30:54.664 properties.go:GetString:28 → folder_separator_template:
[DEBUG] 20:30:54.664 properties.go:GetString:28 → folder_separator_icon: /
[DEBUG] 20:30:54.664 properties.go:GetBool:22 → cycle_folder_separator: false
[DEBUG] 20:30:54.664 properties.go:GetString:28 → folder_format: %s
[DEBUG] 20:30:54.664 properties.go:GetString:28 → edge_format: %s
[DEBUG] 20:30:54.664 properties.go:GetString:28 → left_format: %s
[DEBUG] 20:30:54.664 properties.go:GetString:28 → right_format: %s
[TRACE] 20:30:54.664 terminal.go:Flags() - 42ns
[TRACE] 20:30:54.664 terminal.go:GOOS() - 42ns
[TRACE] 20:30:54.664 terminal.go:StackCount() - 41ns
[DEBUG] 20:30:54.664 text.go:patchTemplate:123 → {{ .Data.Code }}
[TRACE] 20:30:54.664 terminal_unix.go:DirIsWritable(/Users/markohara) - 16.833µs
[DEBUG] 20:30:54.664 text.go:patchTemplate:123 →  {{ .Data.Path }}
[TRACE] 20:30:54.665 text.go:Render( {{ .Path }} ) - 998µs
[TRACE] 20:30:54.665 text.go:Render({{ .Code }}) - 1.546916ms
[TRACE] 20:30:54.665 text.go:Render(plain) - 83ns
[DEBUG] 20:30:54.665 properties.go:GetBool:22 → always_enabled: false
[TRACE] 20:30:54.665 text.go:Render(plain) - 41ns
[TRACE] 20:30:54.665 text.go:Render(plain) - 0s
[TRACE] 20:30:54.665 terminal.go:Shell() - 42ns
[TRACE] 20:30:54.665 terminal.go:Shell() - 42ns
[TRACE] 20:30:54.665 terminal.go:Shell() - 0s
[TRACE] 20:30:54.665 terminal.go:Flags() - 42ns
[DEBUG] 20:30:54.665 home.go:12 → /Users/markohara
[TRACE] 20:30:54.665 path.go:Path() - 15.75µs
[TRACE] 20:30:54.665 terminal.go:Flags() - 0s
```

--- a/src/template/text.go
+++ b/src/template/text.go
@@ -5,6 +5,7 @@ import (
 	"reflect"
 	"strings"
 	"time"
+	"unicode"
 
 	"github.com/jandedobbeleer/oh-my-posh/src/log"
 )
@@ -125,7 +126,7 @@ func (t *Text) patchTemplate() {
 
 type fields map[string]bool
 
-func (f *fields) init(data any) {
+func (f fields) init(data any) {
 	if data == nil {
 		return
 	}
@@ -138,16 +139,16 @@ func (f *fields) init(data any) {
 		// check if we already know the fields of this struct
 		if kf, OK := knownFields.Load(name); OK {
 			for key := range kf.(fields) {
-				(*f)[key] = true
+				f.add(key)
 			}
 			return
 		}
 
 		// Get struct fields and check embedded types
 		fieldsNum := val.NumField()
-		for i := 0; i < fieldsNum; i++ {
+		for i := range fieldsNum {
 			field := val.Field(i)
-			(*f)[field.Name] = true
+			f.add(field.Name)
 
 			// If this is an embedded field, get its methods too
 			if !field.Anonymous {
@@ -165,21 +166,32 @@ func (f *fields) init(data any) {
 		// Get pointer methods
 		ptrType := reflect.PointerTo(val)
 		methodsNum := ptrType.NumMethod()
-		for i := 0; i < methodsNum; i++ {
-			(*f)[ptrType.Method(i).Name] = true
+		for i := range methodsNum {
+			f.add(ptrType.Method(i).Name)
 		}
 
-		knownFields.Store(name, *f)
+		knownFields.Store(name, f)
 	case reflect.Map:
 		m, ok := data.(map[string]any)
 		if !ok {
 			return
 		}
 		for key := range m {
-			(*f)[key] = true
+			f.add(key)
 		}
 	case reflect.Ptr:
 		f.init(reflect.ValueOf(data).Elem().Interface())
+	}
+}
+
+func (f fields) add(field string) {
+	if len(field) == 0 {
+		return
+	}
+
+	r := []rune(field)[0]
+	if unicode.IsUpper(r) {
+		f[field] = true
 	}
 }
 

--- a/src/template/text.go
+++ b/src/template/text.go
@@ -4,6 +4,7 @@ import (
 	"fmt"
 	"reflect"
 	"strings"
+	"sync"
 	"time"
 	"unicode"
 
@@ -31,7 +32,7 @@ func (t *Text) Render() (string, error) {
 }
 
 func (t *Text) patchTemplate() {
-	fields := make(fields)
+	fields := &fields{}
 	fields.init(t.Context)
 
 	var result, property string
@@ -124,11 +125,18 @@ func (t *Text) patchTemplate() {
 	log.Debug(t.Template)
 }
 
-type fields map[string]bool
+type fields struct {
+	values map[string]bool
+	sync.RWMutex
+}
 
-func (f fields) init(data any) {
+func (f *fields) init(data any) {
 	if data == nil {
 		return
+	}
+
+	if f.values == nil {
+		f.values = make(map[string]bool)
 	}
 
 	val := reflect.TypeOf(data)
@@ -138,9 +146,7 @@ func (f fields) init(data any) {
 
 		// check if we already know the fields of this struct
 		if kf, OK := knownFields.Load(name); OK {
-			for key := range kf.(fields) {
-				f.add(key)
-			}
+			f.append(kf)
 			return
 		}
 
@@ -184,24 +190,55 @@ func (f fields) init(data any) {
 	}
 }
 
-func (f fields) add(field string) {
+func (f *fields) append(values any) {
+	if values == nil {
+		return
+	}
+
+	fields, ok := values.(*fields)
+	if !ok {
+		return
+	}
+
+	f.Lock()
+	fields.RLock()
+
+	defer func() {
+		f.Unlock()
+		fields.RUnlock()
+	}()
+
+	for key := range fields.values {
+		f.values[key] = true
+	}
+}
+
+func (f *fields) add(field string) {
 	if len(field) == 0 {
 		return
 	}
 
 	r := []rune(field)[0]
-	if unicode.IsUpper(r) {
-		f[field] = true
+	if !unicode.IsUpper(r) {
+		return
 	}
+
+	f.Lock()
+	defer f.Unlock()
+
+	f.values[field] = true
 }
 
-func (f fields) hasField(field string) bool {
+func (f *fields) hasField(field string) bool {
 	field = strings.TrimPrefix(field, ".")
 
 	// get the first part of the field
 	splitted := strings.Split(field, ".")
 	field = splitted[0]
 
-	_, ok := f[field]
+	f.RLock()
+	defer f.RUnlock()
+
+	_, ok := f.values[field]
 	return ok
 }

